core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java

```diff
@@ -1508,6 +1508,24 @@ public SqlOperandCountRange getOperandCountRange() {
           OperandTypes.CHARACTER,
           SqlFunctionCategory.STRING);
 
+  public static final SqlFunction LEFT =
+      new SqlFunction(
+          "LEFT",
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.VARCHAR_2000,
```

Comment:
The return type of `left` can be nullable.

Reply:
hi @pengzhiwei2018 Thank you very much. It looks like there's a conflict between LEFT semantics and LEFT JOIN semantics.

Reply:
The type inference should be the same as SUBSTRING. It can return null if and only if at least one argument is null.

Reply:
hi @julianhyde Thank you very much.It looks like there's a conflict between String's LEFT semantics and LEFT JOIN semantics.I am not sure how to solve this problem. Could you give me some advice?
best
qianjin

Reply:
Hi @XuQianJin-Stars ,you can define a LeftFunctionCall in `BuiltinFunctionCall()` of the Parser.jj, just like the `TimestampAddFunctionCall()` does.  It works well for me .May you succeed.

Reply:
hi @pengzhiwei2018 I've tried it and still got the same error.

Reply:
hi @pengzhiwei2018 Thank you very much for your help.

Reply:
The return type of left can be nullable. Please fix this problem first.

Reply:
well, let me fix it.